docs: rework mdbook (#1036)
coreos/clair: Documentation/whatis.md

> # What is ClairV4

Reviewer: ```suggestion
# What is Clair
```

> ClairV4 is an application for parsing container contents and reporting any vulnerabilities affecting the image. These actions happen via static analysis and not during container runtime.

Reviewer: ```suggestion
Clair is an application for parsing image contents and reporting vulnerabilities affecting the contents. This is done via static analysis and not at runtime.
```

## Architecture

> ClairV4 utilizes the [ClairCore](https://quay.github.io/claircore/) library as its engine for extracing image contents and reporting vulnerabilities. At a high level you can consider ClairV4 a service wrapper to the functionality provided in the ClairCore library. 

Reviewer: ```suggestion
Clair v4 utilizes the [ClairCore](https://quay.github.io/claircore/) library as its engine for examining contents and reporting vulnerabilities. At a high level you can consider Clair a service wrapper to the functionality provided in the ClairCore library. 
```

![diagram of clairV4 highlevel architecture](./clairv4_arch.png)

> The above diagram expresses the separation of concerns between ClairV4 and the ClairCore library. Most development involving new distrubtion, vulnerability analysis, and container indexing will occur in ClairCore.

Reviewer: ```suggestion
The above diagram expresses the separation of concerns between Clair and the ClairCore library. Most development involving new distributions, vulnerability sources, and layer indexing will occur in ClairCore.
```

> ## How ClairV4 Works

Reviewer: ```suggestion
## How Clair Works
```

> ClairV4's container analysis is broken into three distinct parts.

Reviewer: ```suggestion
Clair's analysis is broken into three distinct parts.
```

### Indexing

> Indexing is the act of submitting a Manifest to ClairV4. On receipt ClairV4 will fetch all the layers, scan each layer for contents, and create an intermediate representation of the contents called an IndexReport. 

Reviewer: > Indexing is the act of submitting a Manifest to ClairV4.

This is inaccurate.

```suggestion
Indexing starts with submitting a Manifest to Clair. On receipt, Clair will fetch layers, scan their contents, and return an intermediate representation called an IndexReport. 
```

> Manifests are treated as content-addressable containeers. A manifest's hash will always represent the same content and ClairV4 exploits this fact, not performing duplicate work unless necessary.

Reviewer: ```suggestion
Manifests are Clair's representation of a container image. Clair leverages the fact that OCI Manifests and Layers are content-addressed to reduce duplicated work.
```

> Once a Manifest is indexed the IndxReport is persisted for later retrieval. 

Reviewer: ```suggestion
Once a Manifest is indexed, the IndexReport is persisted for later retrieval. 
```

### Matching

> Matching is the act of taking an IndexReport and discovering vulnerabilties affecting the container the report represents. 

Reviewer: ```suggestion
Matching is taking an IndexReport and correlating vulnerabilities affecting the manifest the report represents. 
```

> The matcher takes care to not cache any datat for too long. ClairV4 is continually indexing new security data and a request to the matcher will always provide you with the most up to date vulnerability analysse of an IndexReport.

Reviewer: ```suggestion
Clair is continually ingesting new security data and a request to the matcher will always provide you with the most up to date vulnerability analysis of an IndexReport.
```

> *how we implement indexing and matching in detail is covered in [ClairCore's](https://quay.github.io/claircore/) documentation*

Reviewer: ```suggestion
*How we implement indexing and matching in detail is covered in [ClairCore's](https://quay.github.io/claircore/) documentation.*
```

### Notifications

> ClairV4 implements a notification service. 

Reviewer: ```suggestion
Clair implements a notification service. 
```

> When new vulnerabilities are discovered the notifier service will determine if these vulnerabilities affect any indexed Manifests. The notifier will then fire a web hook or deliver a notification to a message broker depending on its configuration. 

Reviewer: ```suggestion
When new vulnerabilities are discovered, the notifier service will determine if these vulnerabilities affect any indexed Manifests. The notifier will then take action according to its configuration.
```

### Getting Started

> At this point you'll probably want to check out [Getting Started With ClairV4](./howto/getting_started.md)

Reviewer: ```suggestion
At this point you'll probably want to check out [Getting Started With Clair](./howto/getting_started.md).
```
